source/material_model/steinberger.cc

```diff
@@ -132,9 +132,53 @@ namespace aspect
     void
     Steinberger<dim>::initialize()
     {
+      // This model allows the user to provide several PerpleX P-T lookup files,
+      // each of which corresponds to a different material.
+      // The thermodynamic properties and stable mineral phases within each material
+      // will in general be different, and must be averaged in a reasonable way.
+
+      // In the following, we populate the unique_phase_names vector.
+      // We do this because we are choosing in this material model to combine
+      // minerals with different compositions but which are the same phase.
+      std::set<std::string> set_phase_volume_column_names;
       for (unsigned i = 0; i < material_file_names.size(); i++)
-        material_lookup.push_back(std_cxx14::make_unique<MaterialModel::MaterialUtilities::Lookup::PerplexReader>
-                                  (data_directory+material_file_names[i],interpolation,this->get_mpi_communicator()));
+        {
+          material_lookup.push_back(std_cxx14::make_unique<MaterialModel::MaterialUtilities::Lookup::PerplexReader>
+                                    (data_directory+material_file_names[i],interpolation,this->get_mpi_communicator()));
+
+          // Here we look up all of the column names and insert
+          // unique names into the unique_phase_names vector.
+          std::vector<std::string> phase_volume_column_names = material_lookup[i]->phase_volume_column_names();
+          for (std::string const &name : phase_volume_column_names)
+            {
+              if (std::find(unique_phase_names.begin(),
+                            unique_phase_names.end(),
+                            name) == unique_phase_names.end())
+                unique_phase_names.push_back(name);
+            }
+        }
+
+      // We now fill the unique_phase_indices object
+      // with the unique_phase_names index for each of the phases
+      // stored in each material lookup.
+
+      // Initialise the column indices
+      unique_phase_indices.resize(material_file_names.size(),
+                                  std::vector<unsigned int>());
```

Comment:
I would put that on one line. Also: you can resize this before you start your first loop in line 144, and then you only need one loop over all i.

Reply:
I don't think this is true. The second loop over i requires `unique_phase_names`, which is only complete at the very end of the first loop over i.

Reply:
The call to resize is now on one line.

Reply:
Oh, wait, now I see what you mean. When this involved a set, I needed the two loops, but now I don't. Very good catch!